Scripts/RemediationScripts/Remediate-EnableHTTPSForAppServices.ps1

```diff
@@ -0,0 +1,917 @@
+        3. Execute the script to disable HTTPS on the production slot and all non-production slots in all App Services in the Subscription. Refer `Examples`, below.
+
+# Examples:
+    To remediate:
+        1. To review the App Services in a Subscription that will be remediated:
+    
+           Enable-HttpsForAppServices -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -DryRun
+
+        2. To enable HTTPS on the production slot and all non-production slots of all App Services in a Subscription:
+       
+           Enable-HttpsForAppServices -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck
+
+        3. To enable HTTPS on the production slot and all non-production slots of all App Services in a Subscription, from a previously taken snapshot:
+       
+           Enable-HttpsForAppServices -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -FilePath C:\AzTS\Subscriptions\00000000-xxxx-0000-xxxx-000000000000\202109131040\EnableHTTPSForAppServices\AppServicesWithoutHTTPSEnabled.csv
+
+        To know more about the options supported by the remediation command, execute:
+        
+        Get-Help Enable-HttpsForAppServices -Detailed
+
+    To roll back:
+        1. To disable HTTPS on the production slot and all non-production slots of all App Services in a Subscription, from a previously taken snapshot:
+           Disable-HttpsForAppServices -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -FilePath C:\AzTS\Subscriptions\00000000-xxxx-0000-xxxx-000000000000\202109131040\EnableHTTPSForAppServices\RemediatedAppServices.csv
+        
+        2. To disable HTTPS on the production slot and all non-production slots of all App Services in a Subscription, from a previously taken snapshot:
+           Disable-HttpsForAppServices -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -ExcludeNonProductionSlots -FilePath C:\AzTS\Subscriptions\00000000-xxxx-0000-xxxx-000000000000\202109131040\EnableHTTPSForAppServices\RemediatedAppServices.csv
+
+        To know more about the options supported by the roll back command, execute:
+        
+        Get-Help Disable-HttpsForAppServices -Detailed        
+###>
+
+
+function Setup-Prerequisites
+{
+    <#
+        .SYNOPSIS
+        Checks if the prerequisites are met, else, sets them up.
+
+        .DESCRIPTION
+        Checks if the prerequisites are met, else, sets them up.
+        Includes installing any required Azure modules.
+
+        .INPUTS
+        None. You cannot pipe objects to Setup-Prerequisites.
+
+        .OUTPUTS
+        None. Setup-Prerequisites does not return anything that can be piped and used as an input to another command.
+
+        .EXAMPLE
+        PS> Setup-Prerequisites
+
+        .LINK
+        None
+    #>
+
+    # List of required modules
+    $requiredModules = @("Az.Accounts", "Azure")
```

Comment:
We might need to add Az.WebApp module.

Reply:
Good catch. Added `Az.Websites` to the list of required modules.